website/tutorial/basic/09-error-handling.mdx

```diff
@@ -0,0 +1,253 @@
+
+## Exposing safe error messages
+
+So let's say instead of the message `Unexpected error.`, we instead want to raise an error with the message `Unauthenticated!`.
+
+The `@graphql-yoga/node` package exports the `GraphQLYogaError` class.
+
+<Instruction>
+
+Instead of throwing an `Error` you will adjust the code within `src/schema.ts` to throw an `GraphQLYogaError` instead.
+
+```typescript
+// ... other imports ...
+import { GraphQLYogaError } from '@graphql-yoga/node'
+// ... other imports ...
+
+const resolvers = {
+  // ... other resolver maps ...
+  Mutation: {
+    // ... other Mutation object type field resolver functions ...
+    post: async (
+      parent: unknown,
+      args: { url: string; description: string },
+      context: GraphQLContext,
+    ) => {
+      // authorization logic
+      if (context.currentUser === null) {
+        throw new GraphQLYogaError('Unauthenticated!')
+      }
+
+      const newLink = await context.prisma.link.create({
+        data: {
+          url: args.url,
+          description: args.description,
+          postedBy: { connect: { id: context.currentUser.id } },
+        },
+      })
+
+      return newLink
+    },
+  },
+}
+```
+
+</Instruction>
+
+<Instruction>
+
+Restart the server using `npm run dev` and execute the mutation operation again without an authorization token.
+
+```graphql
+mutation {
+  post(description: "Google", url: "http://google.de") {
+    id
+  }
+}
+```
+
+</Instruction>
+
+You will receive a response with the `Unauthenticated!` message.
+
+```json
+{
+  "errors": [
+    {
+      "message": "Unauthenticated!",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": ["me"]
+    }
+  ],
+  "data": null
+}
+```
+
+<Instruction>
+
+Restart the server using `npm run start` and execute the mutation operation again without an authorization token.
+
+```graphql
+mutation {
+  post(description: "Google", url: "http://google.de") {
+    id
+  }
+}
+```
+
+</Instruction>
+
+Even without `NODE_ENV` being set to `development` you will receive the same `Unauthenticated!` message.
+
+```json
+{
+  "errors": [
+    {
+      "message": "Unauthenticated!",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": ["me"]
+    }
+  ],
+  "data": null
+}
+```
+
+GraphQLYoga will exclude `GraphQLYogaError` errors being thrown within the resolvers from masking.
+
+That means everytime you want to expose an error to the outside world you should throw a `GraphQLYogaError` error.
+
+Any other unexpected errors will automatically masked from the outside world, bringing you sensible and safe defaults for your future Yoga production deployment!
+
+<Instruction>
+
+You can now replace all remaining `Error` constructor calls with `GraphQLYogaError` constructor calls.
```

Comment:
Should we also link to this post once it's published?
https://github.com/the-guild-org/the-guild-website/pull/611

Reply:
Agree with @n1ru4l, we should point to @n1ru4l article on LogRocket that is easier to understand for beginners

Reply:
added this to the list of https://github.com/dotansimha/graphql-yoga/issues/916

Reply:
> fp-ts is super complex and hard to get into especially if you are coming from this tutori

Only the last part of the post is about fp-ts. There's an important part of the post which is about using codegen to get type safety after having modeled the errors as part of the schema. This is not in notrab's video nor in @n1ru4l article on LogRocket. I think it's an important bit of info if one uses Typescript and not just Javascript.